src/components/Notifications/AccountNotification.jsx

```diff
@@ -25,39 +35,123 @@ const AccountNotification = ({ notificationBlock, today, removeEntry }) => {
+    console.log('calling decline all function');
+
+    const timeId = setTimeout(async () => {
+      await Promise.all(
+        accounts.map(async account => {
+          await account.declineCallback();
+        }),
+      );
+      console.log("set remove notif block true");
+      setRemoveNotificationBlock(true);
+      setDeclineAfterTimer(true);
+    }, 5000); // set 5 sec timer for accept all requests
+    timeoutId=timeId;
+
+    // toast({
+    //   title: `Declined ${accounts?.length} accounts.`,
+    //   status: 'info',
+    //   duration: 9000,
+    //   isClosable: true,
+    // });
+    // removeEntry(notificationBlock.key);
+    console.log('done with decline all function');
+  };
+
+  const undoAll = async () => {
+    console.log('calling undo all function');
+    console.log("clearing timeoutid from undoall");
+    clearTimeout(timeoutId);
+    timeoutId = undefined;
+    console.log("undo called, set remove notif block false");
+    setRemoveNotificationBlock(false);
+    setApproveAfterTimer(false);
+    setDeclineAfterTimer(false);
+    console.log('done undoall function');
   };
 
+  // this is really buggy
+  useEffect(() => {
+    console.log('approveaftertimer or removenoticationblock value changed', approveAfterTimer, removeNotificationBlock);
+    if (approveAfterTimer && removeNotificationBlock) {
+      console.log('approveafter timer = true and remove notification block = true');
+      console.log("REMOVE NOTIF BLOCK HERE");
+
+      toast({
+        title: `Approved.`,
+        status: 'success',
+        duration: 9000,
+        isClosable: true,
+      });
+
+      removeEntry(notificationBlock.key);
+    } else if (declineAfterTimer && removeNotificationBlock) {
+      console.log('decline after timer = true and remove notification block = true');
+      console.log("REMOVE NOTIF BLOCK HERE");
+
+      toast({
+        title: `Declined.`,
+        status: 'info',
+        duration: 9000,
+        isClosable: true,
+      });
+      removeEntry(notificationBlock.key);
+    } else if (approveAfterTimer) {
+      console.log('approveafter timer = true and remove notification block = false');
+      console.log("remove 1 account here");
+      console.log('idToRemove', idToRemove);
+      setAccounts(accounts => accounts.filter(account => account.id !== idToRemove));
+      toast({
+        title: `Approved.`,
+        status: 'success',
+        duration: 9000,
+        isClosable: true,
+      });
+    } else if (declineAfterTimer) {
+      console.log('decline after timer = true and remove notification block = false');
+      console.log("remove 1 account here");
+      console.log('idToRemove', idToRemove);
+      setAccounts(accounts => accounts.filter(account => account.id !== idToRemove));
+      toast({
+        title: `Declined.`,
+        status: 'info',
+        duration: 9000,
+        isClosable: true,
+      });
+    }
+  }, [approveAfterTimer, declineAfterTimer, removeNotificationBlock]);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has missing dependencies: 'idToRemove', 'notificationBlock.key', 'removeEntry', and 'toast'. Either include them or remove the dependency array. If 'removeEntry' changes too often, find the parent component that defines it and wrap that definition in useCallback.